include/ginkgo/core/base/math.hpp

```diff
@@ -1008,6 +1008,24 @@ is_finite(const T &value)
 }
 
 
+/**
+ * Computes the quotient of the given parameters, guarding against division by
+ * zero.
+ *
+ * @tparam T  value type of the parameters
+ *
+ * @param a  the dividend
+ * @param b  the divisor
+ *
+ * @return the value of `a / b` if b is non-zero, zero otherwise.
+ */
+template <typename T>
+GKO_INLINE GKO_ATTRIBUTES T safe_divide(T a, T b)
+{
+    return b == zero<T>() ? zero<T>() : a / b;
```

Comment:
Any votes for leaving `inf` as it is, but doing this only for `0/0`? Maybe that's a separate PR, though.

Reply:
This would allow solvers to introduce `NaN` into vectors, which might not be the best choice? Right now, that is only possible in very limited circumstances (overflows in vector entries or b != 0 but a/b overflows)

Reply:
I think it is to fit the original behavior mostly